From e4d269dae343a4f0e1d4be7c272e3fb7079deec2 Mon Sep 17 00:00:00 2001
Message-Id: <e4d269dae343a4f0e1d4be7c272e3fb7079deec2.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Akash Goel <akash.goel@intel.com>
Date: Tue, 1 Jul 2014 20:22:15 +0530
Subject: [PATCH 294/312] FOR_UPSTREAM [VPG]: drm/i915/vlv: Force the
 alignment to 128KB for Linear Frame buffers.

As per the VLV Bpec, the Graphics address assigned to Linear frame buffers
shall be aligned to 128Kb.

Issue: GMIN-2949
Change-Id: I84eebd788c6b30ee9453764c90d09ad7e7fa2265
Signed-off-by: Akash Goel <akash.goel@intel.com>
Signed-off-by: Mahesh Kumar <mahesh1.kumar@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index bfab9eb..aeac228 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2369,7 +2369,8 @@ intel_pin_and_fence_fb_obj(struct drm_device *dev,
 
 	switch (obj->tiling_mode) {
 	case I915_TILING_NONE:
-		if (IS_BROADWATER(dev) || IS_CRESTLINE(dev))
+		if (IS_BROADWATER(dev) || IS_CRESTLINE(dev) ||
+				IS_VALLEYVIEW(dev))
 			alignment = 128 * 1024;
 		else if (INTEL_INFO(dev)->gen >= 4)
 			alignment = 4 * 1024;
-- 
1.7.9.5

